### ISSUE TITLE
Watcher doesn't detect newly created files.

### ISSUE THREAD

**Issue by sebszocinski** [JeffreyWay/laravel-mix#3003]
-   Laravel Mix Version: 6
-   Node Version 12
-   NPM Version 6
-   OS: Mac

Running `mix watch` and then creating a new file doesn't seem to detect and compile that file until I kill the watcher and start it again.

I'm using pugjs so i'm not sure if its related to the [laravel-mix-pug](https://github.com/matejsvajger/laravel-mix-pug) plugin?

**Comment by extendsLcc** [comment]
> I'm using pugjs so i'm not sure if its related to the [laravel-mix-pug](https://github.com/matejsvajger/laravel-mix-pug) plugin?
Probably not related to the plugin, as I have the same problem with the js files.

With only this code in `webpack.mix.js`
`mix.js( 'resources/js/*.js', 'public/js/bundle.js' );`

The Watcher doesn't detect any new js file in `resources/js/` folder util restart.

**Comment by sebszocinski** [comment]
yeah that sucks, I swear it never used to do this a while back when I was on v4.

**Comment by willvincent** [comment]
Honestly, I think this is a webpack issue, and they seem uninterested in resolving it from the various threads I've looked at.

I have a similar scenario where neither watch nor watch-poll are picking up _anything_ - not changes, new files, nothing. Oddly only on one machine, works fine on my laptop. 

I just whipped up this custom script to use as a watcher that _does_ seem to work reliably, and should run on any event - add/unlink/change/etc:

```js
const _debounce = require('lodash.debounce');
const path = require('node:path');
const { spawn } = require('child_process');

const mixPath = path.resolve(__dirname, '..', 'node_modules/.bin/mix');

const mix = function () {
  const output = spawn(mixPath, ['--no-progress']);
  output.stdout.pipe(process.stdout);
}

chokidar.watch([
  path.resolve(__dirname, '..', 'resources/views/**/*.blade.php'),
  path.resolve(__dirname, '..', 'storage/framework/views/**/*.php'),
  path.resolve(__dirname, '..', 'vendor/**/*.blade.php'),
]).on('all', _debounce(mix, 150));
```
